rust/src/basis.rs

```diff
@@ -94,11 +113,11 @@ impl<'a> Basis<'a> {
                 &mut ptr,
             )
         };
-        Self { ceed, ptr }
+        Self { ptr }
```

Comment:
Why drop the ref to the ceed?

Reply:
I'm trying to reintroduce it with an `Rc`, I think it makes more sense that the objects created by the backend tells the backend to not self destroy rather than the backend telling the objects to not self destroy. But this is a questionnable choice. This makes the borrow checking become runtime instead of compile time.